duplicate_code_detection.py

```diff
@@ -139,7 +174,10 @@ def run(fail_threshold, directories, files, ignore_directories, ignore_files,
             # read file but also recover from encoding errors in source files
             with open(source_code_file, 'r', errors='surrogateescape') as f:
                 # Store source code with the file path as the key
-                source_code[source_code_file] = f.read()
+                content = f.read()
+                if only_code:
```

Comment:
```suggestion
                if only_code and source_code_file.endswith('py'):
```
Limit this to python otherwise you'll get errors

Reply:
#8